Utility/Connect-AzureVM.ps1

```
@@ -27,103 +27,198 @@
+
+  function Configure-AzureWinRMHTTPS {
+  <#
+  .SYNOPSIS
+  Configure WinRM over HTTPS inside an Azure VM.
+  .DESCRIPTION
+  1. Creates a self signed certificate on the Azure VM.
+  2. Creates and executes a custom script extension to enable Win RM over HTTPS and opens 5986 in the Windows Firewall
+  3. Creates a Network Security Rules for the Network Security Group attached the the first NIC attached the the VM allowing inbound traffic on port 5986  
+  #>
+ 
+   
+  Param
+          (
+            [parameter(Mandatory=$true)]
+            [String]$VMName,
+             
+            [parameter(Mandatory=$true)]
+            [String]$ResourceGroupName,      
+ 
+            [parameter()]
+            [String]$DNSName = $env:COMPUTERNAME,
+              
+            [parameter()]
+            [String]$SourceAddressPrefix = "*"
+ 
+          ) 
+ 
+# define a temporary file in the users TEMP directory
+$File = $env:TEMP + "\ConfigureWinRM_HTTPS.ps1"
+
+$String = "param(`$DNSName)" + "`r`n" + "Enable-PSRemoting -Force" + "`r`n" + "New-NetFirewallRule -Name 'WinRM HTTPS' -DisplayName 'WinRM HTTPS' -Enabled True -Profile 'Any' -Action 'Allow' -Direction 'Inbound' -LocalPort 5986 -Protocol 'TCP'" + "`r`n" + "`$thumbprint = (New-SelfSignedCertificate -DnsName `$DNSName -CertStoreLocation Cert:\LocalMachine\My).Thumbprint" + "`r`n" + "`$cmd = `"winrm create winrm/config/Listener?Address=*+Transport=HTTPS @{Hostname=`"`"`$DNSName`"`"; CertificateThumbprint=`"`"`$thumbprint`"`"}`"" + "`r`n" + "cmd.exe /C `$cmd"
+	
+$String | Out-File -FilePath $File -force
+ 
+  
+# Get the VM we need to configure
+$VM = Get-AzureRmVM -ResourceGroupName $ResourceGroupName -Name $VMName
+ 
+# Get storage account name
+$StorageAccountName = $VM.StorageProfile.OsDisk.Vhd.Uri.Split("//")[2].Split('.')[0]
+# get storage account key
+$StorageKey = Get-AzureRmStorageAccountKey -ResourceGroupName $ResourceGroupName -Name $StorageAccountName
+try
+{
+    $Key = $StorageKey.value[0]
+}
+catch
+{
```

may want to give an error here instead of silently failing.